### PR TITLE
Add templates (CoC, issue template, etc) that are common to all providers

### PR DESCRIPTION
**Pull request by paddycarey**
This PR adds a number of files that are common to all of the official providers.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+Hi there,
+
+Thank you for opening an issue. Please note that we try to keep the Terraform issue tracker reserved for bug reports and feature requests. For general usage questions, please see: https://www.terraform.io/community.html.
+
+### Terraform Version
+Run `terraform -v` to show the version. If you are not running the latest version of Terraform, please upgrade because your issue may have already been fixed.
+
+### Affected Resource(s)
+Please list the resources as a list, for example:
+- opc_instance
```
Comment (lskillen) [inline .github/ISSUE_TEMPLATE.md:10]: Maybe list our resources as examples?

File: .github/ISSUE_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+### Actual Behavior
+What actually happened?
+
+### Steps to Reproduce
+Please list the steps required to reproduce the issue, for example:
+1. `terraform apply`
+
+### Important Factoids
+Are there anything atypical about your accounts that we should know? For example: Running in EC2 Classic? Custom version of OpenStack? Tight ACLs?
```
Comment (lskillen) [inline .github/ISSUE_TEMPLATE.md:39]: Something something Cloudsmith. Type of repository. Paid or free account. Etc.

Comment (paddycarey) [reply .github/ISSUE_TEMPLATE.md:39]: The template is standard across all the official providers afaics. I checked >10 and the text is identical, nothing provider specific, even the Google and Azure ones mention EC2-classic like here 🤷

Comment (lskillen) [reply .github/ISSUE_TEMPLATE.md:39]: :shrugops: 😁

**Pull request merged**